hardware_interface/include/hardware_interface/robot_hw.h

```diff
@@ -148,33 +152,72 @@ class RobotHW : public InterfaceManager
     ERROR
   };
 
-  // Return (in realtime) the state of the last doSwitch()
+  /** \brief Return (in realtime) the state of the last doSwitch() */
   virtual SwitchState switchResult() const
   {
     return DONE;
   }
 
-  // Return (in realtime) the state of the last doSwitch() for a given controller
+  /** \brief Return (in realtime) the state of the last doSwitch() for a given controller */
   virtual SwitchState switchResult(const ControllerInfo& /*controller*/) const
   {
     return DONE;
   }
+  /**\}*/
+
+  
+  /** \name Control Loop
+   * 
+   */
+   /**\{*/
 
-  /**
-   * Reads data from the robot HW
+  /** \brief Used to read data from the robot hardware (represented by its interfaces) and populate the raw data.
+   *
+   * The read method is part of the control loop cycle (read, update, write) and used to populate the raw data with 
+   * the state of your robot's hardware resources (joints, sensors, actuators). In your node that handles the control loop,
+   * you should call this method before calling controller_manager::ControllerManager::update and write().
+   * 
+   * The raw data describes regions in memory of your custom robot hardware interface (subclassed from hardware_interface::RobotHW),
+   * which is usually represented by member arrays having names that give semantic meaning (pos, vel, eff) to the registered resources.
+   * The registration of memory regions (assigning pointers) is done via JointStateInterface, for read-only joints,
+   * or JointCommandInterface, for joints that accept commands and provide feedback (read and write), and its subclasses
+   * (eg. VelocityJointInterface) that make use of JointHandle or JointStateHandle through the HardwareResourceManager.
```

Comment:
This seems a little out of place to me. Registering interfaces should probably be up in the documentation for `init()`, since that's where it normally happens. I don't think we need discussion about how to store the robot state in `RobotHW`, since that's pretty trivial and also very much up to the implementation.

Reply:
> Could you please clarify where the registration of interfaces should take place or is it just personal preference?

Off the top of my head, I can't think of a reason it would make a difference whether it's in the constructor or `init`, I've just always tended to put it in `init` :) Unless someone else chimes in with a good reason, I don't think it matters. I also don't think there's need to recommend one way vs the other, can we word the documentation to be inclusive of either option?

> Do you know why the init() method of ros_control_boilerplate isn't using the same parameters as the base class?

That looks like a mistake to me, but let's ping @davetcoleman to know for sure. Looks like a good example of a place where `override` would help.

> Wouldn't it be possible to move the call of `init(ros::NodeHandle& , ros::NodeHandle &)` inside the controller_manager constructor to avoid having the user calling init()?

Perhaps, but that might remove a bit of flexibility in case people are doing funky custom stuff. At this point in the lifecycle of `ros_control`, changing the API without a strong reason doesn't seem like a good idea to me.

Reply:
@bmagyar Can you confirm what I said in my first point above about registering interfaces in the constructor vs init?

> Off the top of my head, I can't think of a reason it would make a difference whether it's in the constructor or init, I've just always tended to put it in init :) Unless someone else chimes in with a good reason, I don't think it matters. I also don't think there's need to recommend one way vs the other, can we word the documentation to be inclusive of either option?

Reply:
I do not recall why it is that way, probably a mistake from me or from the sources that inspired the boilerplate. Please make a PR to fix it :-)